cli/command_restore_entry.go

```diff
@@ -0,0 +1,79 @@
+package cli
+
+import (
+	"context"
+	"math"
+	"path/filepath"
+
+	"github.com/alecthomas/kingpin"
+	"github.com/pkg/errors"
+
+	"github.com/kopia/kopia/fs/localfs"
+	"github.com/kopia/kopia/repo"
+	"github.com/kopia/kopia/snapshot/restore"
+	"github.com/kopia/kopia/snapshot/snapshotfs"
+)
+
+const (
+	restoreEntryCommandHelp = `Expand one or more levels of a directory or file from shallow
```

Comment:
It appears that this is sharing a lot of restore code and flags, could this simple be one command called `restore` that also understands `kopia-entry` arguments:

```
kopia restore --shallow=1 ksomehash /some/output
```

(writes `/some/output/some-file.kopia-entry` and `/some/output/some-subdir.kopia-entry`)

```
kopia restore /some/output/some-file.kopia-entry
```
(replaces `/some/output/some-file.kopia-entry` with `/some/output/some-file`)


```
kopia restore /some/output/some-subdir.kopia-entry --shallow=3 
```
(replaces `/some/output/some-subdir.kopia-entry` with `/some/output/some-subdir/` and restores 3 more levels?)

Reply:
I had something more like this suggestion in 1eb6e25. I thought that your previous suggestion above for the `restore`, `restore-entry` command separation ended up making the code both nicer to read and the command structure easier to explain.

I can change this but I'm concerned that the more complicated control flow in `command_restore.go` will end up being more complicated. WDYT?

Reply:
I had previously had less readable changes to  `command_restore.go` to make it able to do regular restores and what is currently provided by `restore-entry`. After your suggestion above to separate `restore-entry` from `restore`, I thought that the changes to `command_restore.go` ended up cleaner. I am worried that merging `restore-entry` and `restore` will make:

* the `restore` command seem more complicated (and be harder to explain)
* the logic in `command_restore.go` end up more complicated and easier to break.

I will do but am not convinced that this is actually a good idea.

Reply:
Fixed in 96ba13e4